models/Discord/Commands/DiscordCommandMentor.js

```diff
@@ -0,0 +1,44 @@
+const DiscordCommand = require('../DiscordCommand.js');
+
+class DiscordCommandMentor extends DiscordCommand {
+
+    constructor(subsystem) {
+        super("mentor", "Give/remove a mentor's rank", 'mentor', subsystem);
+    }
+
+        onRun(message, permissions, args) {
+        var config = this.subsystem.manager.getSubsystem("Config").config;
+        if (args.length < 2) {
+            message.reply("Usage is `" + config.discord_command_character + "addao <add/remove> [ckey]`");
+            return;
+        }
+        var removing = args[0];
+        var ckey = args[1];
+
+        var dbSubsystem = this.subsystem.manager.getSubsystem("Database");
+        dbSubsystem.pool.getConnection((err, connection) => {
+            connection.query('SELECT * FROM `erro_mentor` WHERE `ckey` = ?', [ckey], (error, results, fields) => {
+                if(results.length > 0) {
+                    message.reply("Player is already a mentor.");
+                    return;
+                }
+            });
+            if(removing === "add") {
```

Comment:
This code will also get ran *BEFORE* the above query completes.

Reply:
mane, how do I do the async stuffs

Reply:
You would have to run the rest of the code inside the query callback... so it would all need to be inside this scope https://github.com/yogstation13/yogbot/pull/152/files#diff-d6cc93564c2f9597bae6b03155e5aba5R24